src/squlearn/util/optree/optree_evaluate.py

```diff
@@ -941,19 +942,11 @@ def change_order(n, reorder_list):
             if i[1] == n:
                 return i[0]
 
-    def maximum_decompose(circ):
-        """Helper function to decompose circuits."""
-        circ_dec = circ.decompose()
-        while circ_dec != circ:
-            circ_dec = circ_dec.decompose()
-            circ = circ.decompose()
-        return circ_dec
-
     if circ_in.num_clbits == 0:
         return circ_in.measure_all(inplace=False)
     else:
         qubits = [i for i in range(circ_in.num_qubits)]
-        circ_in = maximum_decompose(circ_in)
+        circ_in = decompose_to_std(circ_in)  # David: Are all gates until here already decomposed?
```

Comment:
Still, please don't use `TODO`-like comments. Use the PR tools (e.g. the review tool) to comment on code.